src/api/v3/locations.py

```diff
@@ -0,0 +1,68 @@
+from django.http import JsonResponse
+
+from cms.models import POI
+from ..decorators import json_response
+
+
+def transform_poi_translation(poi_translation):
+    """
+    Function to create a JSON from a single poi_translation object.
+
+    :param poi_translation: The poi translation object which should be converted
+    :type poi_translation: ~cms.models.pois.poi_translation.POITranslation
+
+    :return: return data necessary for API
+    :rtype: dict
+    """
+
+    # is there a better way to access this?
+    try:
+        icon_url = poi_translation.icon.url
+    except AttributeError:
+        icon_url = ""
+
+    poi = poi_translation.poi
+    return {
+        "id": poi_translation.id,
+        "url": poi_translation.permalink,
+        "path": poi_translation.slug,
+        "title": poi_translation.title,
+        "modified_gmt": poi_translation.last_updated,
+        "excerpt": poi_translation.short_description,
+        "content": poi_translation.description,
+        "available_languages": poi_translation.available_languages,
+        "thumbnail": icon_url,
+        "location": {
+            "id": poi.id,
+            "name": poi_translation.title,
+            "address": poi.address,
+            "town": poi.city,
+            "state": None,
+            "postcode": poi.postcode,
+            "region": poi.region.slug,
+            "country": poi.country,
+            "latitude": poi.latitude,
+            "longitude": poi.longitude,
+        },
+        "hash": None,
+    }
+
+
+@json_response
+# pylint: disable=unused-argument
+def locations(_, region_slug, language_slug):
+    """
+    List all poi and transform result into JSON
+
+    :return: JSON object according to APIv3 locations endpoint definition
+    :rtype: ~django.http.JsonResponse
+    """
+    result = []
+    for poi in POI.objects.all():
```

Comment:
```suggestion
    for poi in region.pois.filter(archived=False):
```
The endpoint should just return the POIs of the requested region, not all POIs.